onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -211,6 +219,16 @@ def update(self, request, *args, **kwargs):
         username = kwargs.get("user")
         response = super().update(request, *args, **kwargs)
         cache.set(f"{USER_PROFILE_PREFIX}{username}", response.data)
+
+        # send notification on updating user profile
+        send_message(
```

Comment:
We can also move this to the serializer. Since we are sending the messaging for both `create` and `update`, we can override the serializer's `save` method and call `send_message` there. [Example](https://github.com/onaio/onadata/blob/main/onadata/libs/serializers/share_project_serializer.py#L143)

We can differentiate create from update by checking [self.instance](https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L202). When doing an update `self.instance` is not None